### PR TITLE
Updating TI CC3220 PAL port for OTA v3

### PR DESCRIPTION
**Pull request by divekarshubham**
Updating TI CC3200 PAL port for OTA v3

Description
-----------

-     Updated the PAL with the latest Error codes and structures
-     Updated project files
-     Updated cmake files to reflect components needed for OTA
-     Added configuration files to support qualification tests

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: vendors/ti/boards/cc3220_launchpad/aws_demos/config_files/ota_config.h [hunk 1]
```diff
@@ -43,6 +66,11 @@
  */
 #define otaconfigLOG2_FILE_BLOCK_SIZE           10UL
 
+/**
```
Comment (pvyawaha) [inline vendors/ti/boards/cc3220_launchpad/aws_demos/config_files/ota_config.h:69]: The comment is for the number of buffers, please keep it number of blocks requested plus 1

File: vendors/ti/boards/cc3220_launchpad/aws_demos/config_files/ota_demo_config.h [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+#endif
+
+/**
+ * @brief Build version of the firmware.
+ *
+ *        This is used in the OTA demo to set the appFirmwareVersion variable
+ *        that is declared in the ota_appversion32.h file in the OTA library.
+ */
+#ifndef APP_VERSION_BUILD
+    #define APP_VERSION_BUILD    0
```
Comment (pvyawaha) [inline vendors/ti/boards/cc3220_launchpad/aws_demos/config_files/ota_demo_config.h:64]: I think in our documents we mention that the initial version is 0.9.2

File: vendors/ti/boards/cc3220_launchpad/aws_tests/config_files/aws_test_ota_config.h [hunk 3]
```diff
@@ -68,6 +68,30 @@
...
+ */
+#define APP_VERSION_MINOR    9
+
+/**
+ * @brief Build version of the firmware.
+ *
+ *        This is used in the OTA demo to set the appFirmwareVersion variable
+ *        that is declared in the ota_appversion32.h file in the OTA library.
+ */
+#define APP_VERSION_BUILD    0
```
Comment (pvyawaha) [inline vendors/ti/boards/cc3220_launchpad/aws_tests/config_files/aws_test_ota_config.h:93]: 0.9.2 as per comment above

File: vendors/ti/boards/cc3220_launchpad/aws_tests/config_files/ota_config.h [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+ * standard default size of 16Kb, then the maximum limit for this must be
+ * calculated from the maximum data response limit (128 KB from service)
+ * divided by the block size. For example if block size is set as 1 KB then the
+ * maximum number of data blocks that we can request is 128/1 = 128 blocks.
+ * Configure this parameter to this maximum limit or lower based on how many
+ * data blocks response is expected for each data requests.
+ *
+ * @note This must be set to a value larger than zero.
+ */
+#define otaconfigMAX_NUM_BLOCKS_REQUEST         4U
```
Comment (pvyawaha) [inline vendors/ti/boards/cc3220_launchpad/aws_tests/config_files/ota_config.h:130]: Number of blocks requested plus 1

File: vendors/ti/boards/cc3220_launchpad/aws_tests/config_files/ota_config.h [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+ */
+#define otaconfigMAX_NUM_REQUEST_MOMENTUM       32U
+
+/**
+ * @brief The number of data buffers reserved by the OTA agent.
+ *
+ * This configurations parameter sets the maximum number of static data buffers used by
+ * the OTA agent for job and file data blocks received.
+ */
+#define otaconfigMAX_NUM_OTA_DATA_BUFFERS       4U
```
Comment (pvyawaha) [inline vendors/ti/boards/cc3220_launchpad/aws_tests/config_files/ota_config.h:147]: Number of blocks requested plus 1

**Review by pvyawaha — CHANGES_REQUESTED** [review]
Please address the comments.

File: vendors/ti/boards/cc3220_launchpad/aws_tests/config_files/ota_config.h (hunk 5 above)
Comment (divekarshubham) [reply vendors/ti/boards/cc3220_launchpad/aws_tests/config_files/ota_config.h]: Updated in https://github.com/aws/amazon-freertos/pull/3179/commits/1a91157fe268d8068568eebd9fcd04a98e7cd81a

File: vendors/ti/boards/cc3220_launchpad/aws_tests/config_files/aws_test_ota_config.h (hunk 3 above)
Comment (divekarshubham) [reply vendors/ti/boards/cc3220_launchpad/aws_tests/config_files/aws_test_ota_config.h]: Updated in https://github.com/aws/amazon-freertos/pull/3179/commits/1a91157fe268d8068568eebd9fcd04a98e7cd81a

File: vendors/ti/boards/cc3220_launchpad/aws_demos/config_files/ota_demo_config.h (hunk 2 above)
Comment (divekarshubham) [reply vendors/ti/boards/cc3220_launchpad/aws_demos/config_files/ota_demo_config.h]: Updated in https://github.com/aws/amazon-freertos/pull/3179/commits/1a91157fe268d8068568eebd9fcd04a98e7cd81a

**Review by pvyawaha — APPROVED** [review]

**Pull request merged**